lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -144,10 +144,31 @@ function isObjectLocked(bucket, objectMD, headers) {
     return false;
 }
 
+function validateObjectLockUpdate(objectMD, retentionInfo) {
+    const { retentionMode: existingMode, retentionDate: existingDateISO } = objectMD;
+    if (!existingMode || existingMode === 'GOVERNANCE') {
+        return null;
+    }
+
+    const _existingDate = new Date(existingDateISO);
```

Comment:
Now that the other date is a separate name you can remove the underscore :smirk: